core/sql/runtimestats/ssmpipc.cpp

```diff
@@ -540,7 +540,7 @@ static Int64 SikGcInterval = -1;
 void SsmpGlobals::work()
 {
 
-  getIpcEnv()->getAllConnections()->waitOnAll(getStatsMergeTimeout());
+  getIpcEnv()->getAllConnections()->waitOnAll(0);
```

Comment:
We would want ssmp to wake up every 3 seconds if there are no other requests to it and do some cleanup tasks. So, it is not clear why do you want to set this to 0.

Reply:
I have move the wait out to the main loop, so there is no need to wait here.

Reply:
The wait in the main loop will wake up every 3 seconds, so here we only need to check if there is any work to do, and no need to wait.

Reply:
I got it now. My understanding is that we wait for openers or messages from openers in XWAIT. If there are open messages, it is consumed at cc->wait(0) and if there are messages from openers it is consumed here. But we wake up every 3 seconds to do bookkeeping.

Reply:
Yes, that's right